modules/caddyhttp/headers/headers.go

```diff
@@ -138,6 +141,15 @@ func (ops *HeaderOps) Provision(_ caddy.Context) error {
 			}
 		}
 	}
+	for _, delHeader := range ops.Delete {
+		if strings.ContainsAny(delHeader, "+|*|[") {
```

Comment:
I know this may be unlikely, but I'm not sure we can safely assume that header field names are alphanumeric.

I had to look this up in the HTTP/2 spec, which actually defers to the HTTP/1.1 spec for describing acceptable header field values, and that spec says the header field names can be `token`, which I finally found, is defined as:

```
<any CHAR except CTLs or separators>
```

which is basically any ASCII character, including the "special" regex characters.

We also can't assume that a regex _does_ have one of those characters, necessarily. A regex can also be something like `foo`.

I think there'll need to be some sort of other way to indicate that the user is specifying a regular expression.

Reply:
Maybe `-~` prefix? Is that insane? :joy:

Reply:
Maybe, if it's really needed. I'd like to see if a simple prefix match would suffice first, since that'd be fairly intuitive and really really simple.

Reply:
Can we go with `-~` ?

Reply:
Is there an actual use case that needs more than a prefix match?